TemporaryParser-Tests/TPRBParserTest.class.st

```diff
@@ -168,104 +178,437 @@ TPRBParserTest >> testBlockReturnNode [
+]
+
+{ #category : #testCreationProtocol }
+TPRBParserTest >> testCreationMessageNodeFromExpression [
 	| messageNode |
-	self compare: (TPRBMessageNode 
+	"Verify the good parsing of an expression."
+	self compare: (self parserClass parseExpression: 'self + 0')
+		  to: (TPRBMessageNode 
 				receiver: (TPRBVariableNode named: 'self')
 				selector: #+
-				arguments: (Array with: (TPRBLiteralNode value: 0)))
-		to: (self parserClass parseExpression: 'self + 0').
-	messageNode := TPRBMessageNode receiver: (TPRBVariableNode named: 'self') selector: #foo.
-	self compare: (TPRBMethodNode selector: #bar
-				body: (TPRBSequenceNode statements: (OrderedCollection 
-								with: (TPRBCascadeNode messages: (OrderedCollection with: messageNode with: messageNode)))))
-		to: (self parserClass parseMethod: 'bar self foo; foo')
+				arguments: (Array with: (TPRBLiteralNode value: 0))).
+]
+
+{ #category : #testCreationProtocol }
+TPRBParserTest >> testCreationSelfNodeWithExpression [
+	self compare: (self parserClass parseExpression: 'self')
+		  to: (TPRBSelfNode new).
+]
+
+{ #category : #testCreationProtocol }
+TPRBParserTest >> testCreationSuperNodeWithExpression [
+	self compare: (self parserClass parseExpression: 'super')
+		  to: (TPRBSuperNode new).
 ]
 
-{ #category : #'tests comparing' }
-TPRBParserTest >> testEqualToWithMapping [
+{ #category : #testCreationProtocol }
+TPRBParserTest >> testCreationThisContexNodeWithExpression [
+	self compare: (self parserClass parseExpression: 'thisContext')
+		  to: (TPRBThisContextNode new).
+]
+
+{ #category : #testCreationProtocol }
+TPRBParserTest >> testCreationVariableNodeWithExpression [
+	self compare: (self parserClass parseExpression: 'variable')
+		  to: (TPRBVariableNode named: #variable start: 1).
+]
+
+{ #category : #testArray }
+TPRBParserTest >> testDelimitationsOfArrayNodesAreBrackets [
 	| tree |
-	tree := self treeWithEverything.
-	self assert: (tree equalTo: tree withMapping: Dictionary new).
-	tree := self treeWithReallyEverything.
-	self assert: (tree equalTo: tree withMapping: Dictionary new).
+	#('{}' '{.}' '{..}' '{foo. bar}' '{foo. bar.}' '{foo. bar. .}' '{. foo. bar}' '{foo.. bar}')
+	do: [ :each |  tree := self parserClass parseExpression: each.
+						self assert: tree left equals: 1.
+						self assert: tree right equals: each size ]
+]
+
+{ #category : #'error testing' }
+TPRBParserTest >> testFaultyLiteralRaiseSyntaxError [
+	#('#"bar"foo' '# foo' '#1' '#12' '#12.3' '# 1' '##1' '#"bar"1') 
+		do: [ :expr | 
+		self should: [self parserClass parseExpression: expr] raise: SyntaxErrorNotification ]
+]
+
+{ #category : #'garbage tests' }
+TPRBParserTest >> testGarbageAssignementOfANonVariable [
+	"Raising a syntax exception when trying to do an assignement on a message."
+	self should: [self parserClass parseExpression: 'receiver selector := 1 + 2'] raise: SyntaxErrorNotification.
+]
+
+{ #category : #'garbage tests' }
+TPRBParserTest >> testGarbageAssignementWithMisplacedDot [

```

Comment:
I'd call it `testNonPresentAssignmentRightValueRaisesSyntaxError`.
I think we should use use the terminology **left value**, and **right value** for assignments. Everybody uses that in parsers/compilers, so it should be understandable :).

Reply:
Le but de ce test est de proposer une approche lorsqu'une ponctuation est malencontreusement insérée dans un assignement. La présence de différents types d'erreurs est donc intentionnel. Il s'agit Syntaxiquement de problèmes différents mais dans la pratique, il s'agit du même problème d'inattention de la part de l'utilisateur.